forge/src/runner.rs

```diff
@@ -258,7 +258,8 @@ impl<'a, S: Clone, E: Evm<S>> ContractRunner<'a, S, E> {
                 }
             },
         };
-        let success = self.evm.check_success(self.address, &status, should_fail);
+        let success = self.evm.check_success(self.address, &status, should_fail) && self.evm.expected_revert().is_none();
```

Comment:
I'd pull in the `self.evm.expected_revert().is_none()` check inside `check_success`

Reply:
nice! test is passing now :)